### PR TITLE
feat: Update to useRouter from next/navigation

### PR DESCRIPTION
**Pull request by andreisocaciu**
Implemented the same functionality as before, but using the new Next.js 13.4 hooks from `'next/navigation'`.

It only works in the `app` directory now, so this means it should be released under v2.0 because of this breaking change.

Closes: #327

### CODE REVIEW

**Comment by franky47** [comment]
Would there be a way to import this implementation in parallel to the previous one, using different import paths?

Eg:
```ts
// Using pages router
import { useQueryState } from 'next-usequerystate' // kept for compatibility
import { useQueryState } from 'next-usequerystate/pages' // explicit "legacy" import

// Using app router
import { useQueryState } from 'next-usequerystate/app'
```

**Comment by franky47** [comment]
@andreisocaciu I tried updating the deps in https://github.com/47ng/next-usequerystate/commit/c46254a7, but somehow it didn't end up in this PR. Could you try replicating those changes and I'll approve the CI runs.

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ jobs:
       - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
       - uses: actions/setup-node@56899e050abffc08c2b3b61f3ec6a79a9dc3223d
```
Comment (franky47) [inline .github/workflows/cd.yml:18]: Sorry about that, I realised that the `actions/setup-node` must also be bumped to v3:

```yml
- uses: actions/setup-node@v3
  with:
    node-version: lts/*
```

**Review by franky47 — CHANGES_REQUESTED** [review]

**Comment by MaxLeiter** [comment]
FWIW `next/navigation` works in both pages and app. The types are different depending on if a `pages` folder is present -- in `pages` the new hooks can return `null` when `router.isReady` would be false.

**Comment by simonexmachina** [comment]
Hello, I'm keen to use this in my project. Wondering if this will be merged soon, or if there's another fork that I can use?

**Comment by luisanton-io** [comment]
This would be very helpful, hopefully merged soon...?

**Comment by iloveitaly** [comment]
Same here!

**Review by franky47 — APPROVED** [review]

**Comment by franky47** [comment]
I've merged these changes into `next` (somehow the PR got stuck in conflict limbo..), with some changes to make it possible to use both the app and pages router, using different imports:
```ts
import {} from 'next-usequerystate' // uses the pages router by default for compatibility
import {} from 'next-usequerystate/pages' // explicit opt-in to pages router
import {} from 'next-usequerystate/app' // explicit opt-in to app router
```

I published `1.8.0-beta.1` to test it on my website that uses the app router, feel free to test that version and report back!

**Comment by tryanmac** [comment]
Works like a dream for me thanks all contributors!

**Pull request closed without merging**